src/otx/backend/native/models/classification/factory.py

```diff
@@ -0,0 +1,289 @@
+# mypy: disable_error_code=misc
+
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Module for OTX classification factory."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, overload
+
+from otx.backend.native.models.base import DefaultOptimizerCallable, DefaultSchedulerCallable
+from otx.metrics.accuracy import MultiClassClsMetricCallable
+
+from .hlabel_models import (
+    EfficientNetHLabelCls,
+    MobileNetV3HLabelCls,
+    TimmModelHLabelCls,
+    TVModelHLabelCls,
+    VisionTransformerHLabelCls,
+)
+from .multiclass_models import (
+    EfficientNetMulticlassCls,
+    MobileNetV3MulticlassCls,
+    TimmModelMulticlassCls,
+    TVModelMulticlassCls,
+    VisionTransformerMulticlassCls,
+)
+from .multilabel_models import (
+    EfficientNetMultilabelCls,
+    MobileNetV3MultilabelCls,
+    TimmModelMultilabelCls,
+    TVModelMultilabelCls,
+    VisionTransformerMultilabelCls,
+)
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+
+    from otx.backend.native.models.base import DataInputParams
+    from otx.backend.native.schedulers import LRSchedulerListCallable
+    from otx.metrics import MetricCallable
+    from otx.types.label import LabelInfoTypes
+
+
+class MobileNetV3:
+    """Factory class for MobileNetV3 models.
+
+    Args:
+        label_info (LabelInfoTypes): The label information.
+        data_input_params (DataInputParams): The data input parameters such as input size and normalization.
+        freeze_backbone (bool, optional): Whether to freeze the backbone during training. Defaults to False.
+            Note: only multiclass classification supports this argument.
+        model_name (str, optional): The model name. Defaults to "mobilenetv3_large".
+        task (Literal["multi_class", "multi_label", "h_label"], optional): The task type.
+            Can be "multi_class", "multi_label", or "h_label". Defaults to "multi_class".
+        optimizer (OptimizerCallable, optional): The optimizer callable. Defaults to DefaultOptimizerCallable.
+        scheduler (LRSchedulerCallable | LRSchedulerListCallable, optional): The learning rate scheduler callable.
+            Defaults to DefaultSchedulerCallable.
+        metric (MetricCallable, optional): The metric callable. Defaults to MultiClassClsMetricCallable.
+        torch_compile (bool, optional): Whether to compile the model using TorchScript. Defaults to False.
+    """
+
+    @overload
+    def __new__(
+        cls,
+        label_info: LabelInfoTypes,
+        data_input_params: DataInputParams,
+        task: Literal["multi_class", "multi_label", "h_label"] = "multi_class",
+        freeze_backbone: bool = False,
+        model_name: str = "mobilenetv3_large",
```

Comment:
I think we could consider using `Literal` types to explicitly specify the available MobileNet variants instead of a generic `str`.

For MobileNet, there's a limited set of variants:
- **OTX Native MobileNetV3**: `"mobilenetv3_small"`, `"mobilenetv3_large"`
- **TorchVision MobileNet**: `"mobilenet_v3_small"`, `"mobilenet_v3_large"`

Reply:
Done for all factory models and base models. (Mobilenet, EffNet, VIT)